proposed/activity-improvements.md

```diff
@@ -0,0 +1,326 @@
+# Improve Activity API usability and OpenTelemetry integration
+
+\[Github issue](https://github.com/dotnet/runtime/issues/31373) has past discussion.
+
+**PM** @shirhatti, anksr
+
+**Dev** @tarekgh
+
+
+
+ .NET has long had [System.Diagnostics.Activity](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activity?view=netcore-3.1) and [System.Diagnostics.DiagnosticListener](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.diagnosticlistener?view=netcore-3.1) to support [distributed tracing](https://docs.microsoft.com/en-us/azure/azure-monitor/app/distributed-tracing) scenarios. Code that receives, processes, and transmits requests creates Activity objects which have correlation ids. These Activities are published via DiagnosticListener to telemetry monitoring agents such as Application Insights running in the same process. The telemetry agent can then log the information to a remote store where it can be aggregated and visualized to help developers understand the overall flow of work in a distributed system.
+
+Today distributed tracing instrumentation is predominantly only added to a few critical libraries such as ASP.Net's Kestrel web server or the HttpClient implementation in the BCL. Even if other libraries or app developers wanted to add this instrumentation to their code, the current design makes it difficult - we want to fix that.  We have also been watching the [OpenTelemetry](https://opentelemetry.io/) project which is defining a multi-language multi-vendor standard for how distributed tracing should work. Our goal is to ensure that .Net integrates smoothly. For app developers this means that you can add a reference to the OpenTelemetry NuGet package and all the Activity-based instrumentation in your codebase, your 3rd party libraries, and the runtime will light up and get collected. For library authors this means that you can write to a platform standard API that doesn't force any specific dependency on the library users, but lights up in a predictable way when OpenTelemetry (or any other agent) is present.  
```

Comment:
> We have also been watching the [OpenTelemetry](https://opentelemetry.io/) project

"watching"?  Or participating in?

Reply:
From a Microsoft-wide perspective 'participating in' is the more accurate term, thanks : )

Reply:
Leading is even more accurate:

https://github.com/open-telemetry/opentelemetry-dotnet/graphs/contributors
<img width="748" alt="image" src="https://user-images.githubusercontent.com/1633368/76917918-9e1ed480-68a3-11ea-8a97-8228434979d4.png">

https://github.com/open-telemetry/opentelemetry-specification/graphs/contributors
<img width="373" alt="image" src="https://user-images.githubusercontent.com/1633368/76917943-b2fb6800-68a3-11ea-8a21-42fcc7281c29.png">

Reply:
Azure folks are very much participating:

https://github.com/open-telemetry/opentelemetry-dotnet/graphs/contributors
<img width="748" alt="image" src="https://user-images.githubusercontent.com/1633368/76917918-9e1ed480-68a3-11ea-8a97-8228434979d4.png">

https://github.com/open-telemetry/opentelemetry-specification/graphs/contributors
<img width="373" alt="image" src="https://user-images.githubusercontent.com/1633368/76917943-b2fb6800-68a3-11ea-8a21-42fcc7281c29.png">